src/LumiDirectPC_geo.cpp

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2022 Anna Kowalewska
+
+#include "DD4hep/DetFactoryHelper.h"
+#include "DD4hep/Printout.h"
+#include "DD4hep/Shapes.h"
+#include "DDRec/DetectorData.h"
+#include "DDRec/Surface.h"
+#include "XML/Layering.h"
+#include <XML/Helper.h>
+
+using namespace std;
+using namespace dd4hep;
+
+static Ref_t create_detector(Detector& desc, xml_h e, SensitiveDetector sens)
+{
+  sens.setType("calorimeter");
+
+  xml_det_t     x_det           = e;
+  xml_comp_t    x_dim           = x_det.dimensions();
+  xml_comp_t    x_pos           = x_det.position();
+  xml_comp_t    x_rot           = x_det.rotation();
+  //
+  string        det_name        = x_det.nameStr();
+  string        mat_name        = dd4hep::getAttrOrDefault<string>( x_det, _U(material), "PbWO4" );
+  int		det_ID		=x_det.id();
+  DetElement det(det_name, det_ID);
+  //
+  double        sizeX           = x_dim.x();
+  double        sizeY           = x_dim.y();
+  double        sizeZ           = x_dim.z();
+  double        posX            = x_pos.x();
+  double        posY            = x_pos.y();
+  double        posZ            = x_pos.z();
+  double        rotX            = x_rot.x();
+  double        rotY            = x_rot.y();
+  double        rotZ            = x_rot.z();
+
+  Box box( sizeX, sizeY, sizeZ );
+  Volume vol( det_name + "_vol", box, desc.material( mat_name ) );
+  //vol.setVisAttributes( x_det.visStr() );
+  vol.setVisAttributes(desc.visAttributes(x_det.visStr()));
+  vol.setSensitiveDetector(sens);
+
+  Transform3D  pos( RotationZYX(rotX, rotY, rotZ), Position(posX, posY, posZ) );
+
+  Volume motherVol = desc.pickMotherVolume( det );
+  PlacedVolume phv = motherVol.placeVolume( vol, pos );
+  phv.addPhysVolID("system", det_ID);
```

Comment:
Where do you set `sector` and `module`? Do you not need them (is this detector not segmented; is it simply a single block of PbWO4)? If not, then remove from the readout pattern in the xml file.

Reply:
Yes, it is simply a single block of PbW04 detector. I have just removed sector and module part from the readout pattern in the xml file.

Reply:
Thanks.